src/components/movie-list-page/movie-list-page.tsx

```diff
@@ -42,16 +44,20 @@ const genres = ['Documentary', 'Comedy', 'Horror', 'Crime'];
 
 
 
-export default function MovieListPage() {
+export default function MovieListPage(props: any) {
+    const navigate = useNavigate()
+
+    const { movieId } = useParams();
+    const [searchParams, setSearchParams] = useSearchParams();
+
     const [movies, setMovies] = useState<MovieData[]>([]);
-    const [sortBy, setSortBy] = useState('');
-    const [activeGenres, setActiveGenres] = useState<string[]>([...genres]);
-    const [searchText, setSearchText] = useState('');
+    const [sortBy, setSortBy] = useState(searchParams.get('sortBy') || '');
+    const [activeGenres, setActiveGenres] = useState<string[]>(searchParams.get('genre')?.split(',') || []);
+    const [searchText, setSearchText] = useState(searchParams.get('query') || '');
```

Comment:
why we need useState if you already have all the data in searchParams ?

Reply:
reimplemented without state